### PR TITLE
11 organize the code

### LINKED ISSUE #11: Organize the Code
Currently , there is a lot of redundant code. Organize it better with OOPs and methods.

### PR DESCRIPTION
**Pull request by arhaankk**
Updated all files with class-based file structure and OOPs

@Appearls please have a look at main.py for the enhanced combination of all files. thanks :)

### CODE REVIEW

File: classifier.py [hunk 1]
```diff
@@ -1,103 +1,138 @@
...
-num_classes = len(label_encoder.classes_)
+    def build_model(self, embedding_dim=32, hidden_units=64):
+        """
+        Builds and compiles the LSTM-based classifier model.
+        """
+        vocab_size = len(self.tokenizer.word_index) + 1
+        num_classes = len(self.label_encoder.classes_)
+        self.model = Sequential([
+            Embedding(input_dim=vocab_size, output_dim=embedding_dim, input_length=self.max_seq_length),
+            LSTM(hidden_units),
```
Comment (Appearls) [inline classifier.py:86]: Why was the Bidirectional LTSM layer removed?

File: classifier.py [hunk 2]
```diff
@@ -1,103 +1,138 @@
...
+        """
+        Cleans the input text, tokenizes it, pads the sequence, and predicts the language.
+        """
+        cleaned_text = clean_text(text)
+        seq = self.tokenizer.texts_to_sequences([cleaned_text])
+        padded = pad_sequences(seq, maxlen=self.max_seq_length, padding='post')
+        pred = self.model.predict(padded)
+        label = self.label_encoder.inverse_transform([np.argmax(pred)])
+        return label[0]
+if __name__ == "__main__":
```
Comment (Appearls) [inline classifier.py]: Currently, the classifier can only be trained or loaded from this file. Wouldn’t it be better to have a method for that inside the class?

File: classifier.py (hunk 1 above)
Comment (Appearls) [inline classifier.py:84]: The current model is not functioning as expected. As it doesn’t seem to have the layers used in the updated branch

File: transliterator.py [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+            batch_size=batch_size,
+            callbacks=callbacks
+        )
+        return self.model
+
+    def load_existing_model(self):
+        """
+        Loads an existing model using the shared function.
+        """
+        self.model = load_existing_model(self.model_path)
```
Comment (Appearls) [inline transliterator.py:168]: The training history returned by model.fit() isn’t being stored here. In the earlier version, it was stored in a variable, which allowed for analysis or visualization of accuracy and loss over epochs—useful for debugging and tuning. Would it be helpful to retain that?

File: transliterator.py [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+    def preprocess_data(self):
+        """
+        Runs the full preprocessing pipeline:
+          1. Loads the dataset.
+          2. Filters out empty rows.
+          3. Ensures fields are strings.
+          4. Cleans both source and target text using language-specific allowed tokens.
+        """
+        self.load_dataset()
+        self.data = self.data.filter(lambda ex: ex['target'] and ex['source'] and ex['target'].strip() and ex['source'].strip())
```
Comment (Appearls) [inline transliterator.py:72]: Any reason for changing the defined functions to lambdas? Just wondering if it might make future changes or extensions a bit harder.

File: classifier.py (hunk 2 above)
Comment (arhaankk) [reply classifier.py]: fixed

File: transliterator.py (hunk 3 above)
Comment (arhaankk) [reply transliterator.py:168]: fixed

File: transliterator.py (hunk 4 above)
Comment (arhaankk) [reply transliterator.py:72]: its to make it more concise else we would have to make additional 3 functions more in the class. Hence, keeping it concise.

@Appearls
P.S. -> feel free to change it in another issue.

**Comment by arhaankk** [comment]
fixes #19
